src/RegistrationMsg.chpl

```diff
@@ -96,6 +99,199 @@ module RegistrationMsg
         }
     }
 
+    /* 
+    Compile the component parts of a Categorical attach message 
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg name: name of SymTab element
+    :type name: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc attachCategoricalMsg(cmd: string, name: string, 
+                                            st: borrowed SymTab): MsgTuple throws {
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting Categorical components for '%s'".format(cmd, name));
+
+        var repMsg: string;
+                
+        var cats = st.attrib("%s.categories".format(name));
+        var codes = st.attrib("%s.codes".format(name));
+
+        if (cats.startsWith("Error:")) { 
+            var errorMsg = cats;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+        if (codes.startsWith("Error:")) { 
+            var errorMsg = codes;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        repMsg = "categorical;created %s".format(cats);
+        // Check if the categories is numeric or string, if string add byte size
+        if (isStringAttrib(cats)) {
+            var s = getSegString("%s.categories".format(name), st);
+            repMsg += "+created bytes.size %t".format(s.nBytes);
+        }
+
+        repMsg += ";created %s".format(codes);
+
+        // Optional components of categorical
+        if st.contains("%s.permutation".format(name)) {
+            var perm = st.attrib("%s.permutation".format(name));
+            if (perm.startsWith("Error:")) { 
+                var errorMsg = perm;
+                regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR); 
+            }
+            repMsg += ";created %s".format(perm);
```

Comment:
replace `;` with `+`

Reply:
Updated to use `+` delimited string in the response message